### PR TITLE
feat: add new CYPRESS_SCREENSHOT_OPTIONS and exactName

### PR DESCRIPTION
**Pull request by kien-ht**
- BREAKING CHANGE: new `compareSnapshot` arguments type
- BREAKING CHANGE: rename/simplify import path
- BREAKING CHANGE: rename default screenshots and report folder

This pull request triggers a major release, which is to v.2.0.0

## BREAKING CHANGE: new `compareSnapshot` arguments type
```
// Then
cy.compareSnapshot('header')
cy.compareSnapshot('header', 0.2)
cy.compareSnapshot('header', 0.2, { limit: 6, delay: 1000 })

// Now
cy.compareSnapshot('header')
cy.compareSnapshot({
  name: 'header',
  testThreshold: 0.2
})
cy.compareSnapshot({
  name: 'header',
  testThreshold: 0.2,
  recurseOptions: { limit: 6, delay: 1000 }
})
cy.compareSnapshot({
  name: 'header',
  testThreshold: 0.2,
  recurseOptions: { limit: 6, delay: 1000 },
  // new custom cypress screenshot options
  cypressScreenshotOptions: {
    clip: { x: 0, y: 0, width: 100, height: 100 }
  }
  // new exactName option
  exactName: true
})

// Cypress screenshot options can be set globally in the config file
// cypress-image-diff.config.js
const config = {
  CYPRESS_SCREENSHOT_OPTIONS: { disableTimersAndAnimations: false },
  // ... other configs
};

module.exports = config;

```

## BREAKING CHANGE: rename/simplify import path
```diff
-import getCompareSnapshotsPlugin from 'cypress-image-diff-js/dist/plugin';
+import getCompareSnapshotsPlugin from 'cypress-image-diff-js/plugin';

-import compareSnapshotCommand from 'cypress-image-diff-js/dist/command';
+import compareSnapshotCommand from 'cypress-image-diff-js/command';
```

## BREAKING CHANGE: rename default screenshots and report folder
```diff
-cypress-visual-screenshots
+cypress-image-diff-screenshots

-cypress-visual-report
+cypress-image-diff-html-report
```

## New config options:
- make screenshots and report folder configurable
```
// cypress-image-diff.config.js
const config = {
  REPORT_DIR: 'html-report',
  SCREENSHOTS_DIR: 'visual-screenshots'
};

module.exports = config;
```

## New UI Report:
- [The new UI report](https://github.com/kien-ht/cypress-image-diff-html-report) is ready to integrate and will also come along with this pull request. It doesn't require any actual changes in the code other than just documentation (Gitbook). I'll update the documentation accordingly once this pull request has been merged.

image

This pull request fixes:
- #181 
- #187 
- #138 
- #123 
- #131 
- #141

### CODE REVIEW

**Comment by PippoRaimondi** [comment]
Nice one, will review this weekend!

**Review by PippoRaimondi — APPROVED** [review]

**Review by tamasmagyar — APPROVED** [review]
lgtm

**Comment by HubertArciszewski95** [comment]
Hey, when we can expect the merge and v2 of this plugin?
Waiting for v2 eagerly :)

**Comment by kien-ht** [comment]
> Hey, when we can expect the merge and v2 of this plugin? Waiting for v2 eagerly :)

Hi, we're trying our best to resolve this pull request within this week. Stay tuned!

File: src/reporter/template.hbs [hunk 1]
```diff
@@ -544,19 +544,19 @@
...
               <img
-                src="../cypress-visual-screenshots/diff/$__id__.png"
+                src="../cypress-image-diff-screenshots/diff/$__id__.png"
                 alt="$__id__"
               />
             </div>
             <div class="comparison-container">
               <img
-                src="../cypress-visual-screenshots/comparison/$__id__.png"
+                src="../cypress-image-diff-screenshots/comparison/$__id__.png"
```
Comment (tamasmagyar) [inline src/reporter/template.hbs:559]: Does the template work with custom SCREENSHOTS_DIR variable?

If I understand correctly we can set the screenshot folder with SCREENSHOTS_DIR variable, but not sure if the template processes SCREENSHOTS_DIR

Comment (kien-ht) [reply src/reporter/template.hbs:559]: I don't think we should continue to add new features to the old report. That means the `SCREENSHOTS_DIR` and `REPORT_DIR` do not have any effect on the old report, which continues to function with the hardcoded `cypress-image-diff-screenshots` and `cypress-image-diff-html-report` directory. I'll note this in our documentation.

**Comment by PippoRaimondi** [comment]
Just got free'd up, was at web summit... will have a look at this today/this weekend and get it going!

**Comment by PippoRaimondi** [comment]
Almost done with testing! Amazing job! Vaaamos! 🚀

**Review by PippoRaimondi — APPROVED** [review]

**Pull request merged**